angular/projects/admin-nrpti/src/app/mines/mines-list/mines-list.component.ts

```diff
@@ -0,0 +1,210 @@
+import { Location } from '@angular/common';
+import { Component, OnDestroy, OnInit, ChangeDetectorRef } from '@angular/core';
+import { ActivatedRoute, Router, Params } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import {
+  IColumnObject,
+  IPageSizePickerOption,
+  ITableMessage,
+  LoadingScreenService,
+  TableObject,
+  TableTemplateUtils,
+  Utils
+} from 'nrpti-angular-components';
+import { MinesTableRowComponent } from '../mines-rows/mines-table-row.component';
+
+/**
+ * Mine list page component.
+ *
+ * @export
+ * @class MinesListComponent
+ * @implements {OnInit}
+ * @implements {OnDestroy}
+ */
+@Component({
+  selector: 'app-mines-list',
+  templateUrl: './mines-list.component.html',
+  styleUrls: ['./mines-list.component.scss']
+})
+export class MinesListComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public tableData: TableObject = new TableObject({ component: MinesTableRowComponent });
+  public tableColumns: IColumnObject[] = [
+    {
+      name: 'Name',
+      value: 'name',
+      width: 'col-3'
+    },
+    {
+      name: 'owner',
```

Comment:
Capitalize

Reply:
Capitalize: will fix.

The sorting by name is working (though I will change it to `+name` instead of `-name`), it is just the UI column arrow that isn't working.

The parameters in the url don't show up initially, but do appear once you click any of the pagination controls.

It looks like the problem is that any parameters the resolver uses, but which aren't in the url already (IE: default params), don't get propagated to the list component.  So the data resolves with filters/sorting applied as we expect, but the component has no knowledge of it, and so it loads with no params in the url, and no UI indications (like sorting arrows).

So these issues might warrant their own ticket, as we probably want to figure out the best way to do this, and then update all of our list components/resolvers.

> 
> 
> > > There's some model changes indicated in the [PR 393](https://github.com/bcgov/NRPTI/pull/393). Specifically: https://github.com/bcgov/NRPTI/pull/393/files#diff-d33c850871eaa41feb49cec3d912d8d5
> > > Not sure about the schema naming, but if we're changing `owner` to `permitee` should probably change the header/ref in the table when they merge in.
> > 
> > 
> > Yeah for sure. This is where order of merges makes it a bit tricky though. I could per-emptively update this in my PR, but then technically it would be broken, if only temporarily. But also, we won't have any data to load either, so not sure :/
> 
> We don't want to be messing with 'orders' of merges. If a PR can't go in on it's own, then it shouldn't land period, and it should not be submitted for review. There's nothing wrong with an iterative state of the code as long as it doesn't break anything or cause other issues down the road.

Yeah exactly.  I was going to leave it, and then just sync up with Kit about one of us updating it, depending on whos stuff goes in first.

Reply:
> Capitalize
Updated

> Mines list page isn't sorting by name by default, and isn't setting the parameters for the URL properly. All the various pagination and table sorting should be showing up in the URL.
Updated to account for initial/default url parameters.